### ISSUE TITLE
Some Photos broken after autoupload (3.0.6.6)

### ISSUE THREAD

**Issue by jacotec** [nextcloud/ios#1405]
### Expected behaviour

New photos on the iPhone should be automatically and correctly be uploaded if autoupload is configured.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

### Actual behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               

Sometimes autouploaded photos are broken in media preview, the also do not work in the nextcloud browser and are broken on PC's where they sync to.

IMG_1456 and IMG_1457 are both taken within a few seconds. The iPhone was all the time in the home WiFi. They have been auto-uploaded, and the result in the nextcloud view in the browser shows IMG_1456 with 0 bytes:

![image](https://user-images.githubusercontent.com/3321655/91352782-a08a4080-e7ea-11ea-94bf-ed9404401756.png)

The same view in the Nextcloud app on the iPhone shows the pic with 1.9MB, but also broken:

![image](https://user-images.githubusercontent.com/3321655/91352863-c0b9ff80-e7ea-11ea-98ae-7d2d62094ab3.png)

After deleting the file and manually uploading the picture from the iPhone to Nextcloud it works and shows a real size of 2.4MB:

![image](https://user-images.githubusercontent.com/3321655/91353073-08d92200-e7eb-11ea-9d6b-62ebaf6afae4.png)


### Steps to reproduce     

Just having autoupload enabled. Some photos break with the described behavior. I saw if the first time in the 3.0.6 testflight branch. App is set to convert the photos to JPG.

### Reasoning or why should it be changed/implemented?

Photos should always be correctly uploaded. If the upload does not work or does not complete it must be repeated and not leave a broken (partial?) file on the server ... and it may not tag the picture as uploaded in the database of the nextcloud app in the iPhone which prevents the picture from being uploaded again.

### iOS version

13.6

### App version

3.0.6.6

### Server configuration

**Operating system:**

Ubuntu 18.04 LTS

**Web server:**

Apache

**Database:**

MariaDB

**PHP version:**

7.3 (FPM)

**Nextcloud version:** (see Nextcloud admin page)

19.0.1

**Comment by Subito** [comment]
This is probably related to https://github.com/nextcloud/server/issues/7995. If files get uploaded with a client which uses chunked transfers, Nextcloud will sometimes not store the file correctly. This happens with all clients I've tried so far. If you are handling sensitive or important data with your nextcloud, I'd suggest you to take frequent backups (these corruptions can happen on file modifications, too) and create a cronjob which periodicly checks for corrupted files.

**Comment by marinofaggiana** [comment]
@Subito Nextcloud iOS do not use the chunked ... but maybe,  @jacotec you use also Android ?

**Comment by Subito** [comment]
I saw the same bug appear when I tried to upload 20k of pictures with the iOS client to my nextcloud instance. It corrupted about 10% of the pictures. Some additional reporting is here: https://github.com/nextcloud/server/issues/3056. Are you sure th iOS App does not use the `Transfer-Encoding: chunked` Header with `PUT` Requests? Because that was my only explanation on why my photos got corrupted - because it fit the description in the bugs mentioned above. https://github.com/nextcloud/ios/issues/1395 Seems to be the same problem?

**Comment by jacotec** [comment]
> @Subito Nextcloud iOS do not use the chunked ... but maybe, @jacotec you use also Android ?

Nope, only iDevices here 🙂

**Comment by dffvb** [comment]
I have the same problem. Constantly. 5 out of 9 pictures are broken. Thats a joke... data corruption is a no no.

**Comment by jacotec** [comment]
Found two today as well.
Interesting: In the file view of the autouploading device itself, the files do have some size (1.x MB) shown. On all other devices and the web interface these files are shown with 0 Bytes.

**Comment by marinofaggiana** [comment]
Try with the last 3.0.7 build 3 in TestFlight

**Comment by jacotec** [comment]
Just updated a minute ago ...

**Comment by dffvb** [comment]
thanks for the quick response. I have just installed Testflight, and it looks good so far 👍

**Comment by marinofaggiana** [comment]
you have to look at the logs otherwise find the issue is impossible 👌🏻

**Comment by dffvb** [comment]
Ah sorry, already deleted the regular app. I had a look a couple of days, but couldnt find anything suspicious. If it happens again,  I will post the log here.

**Comment by marinofaggiana** [comment]
In the version 3.0.7 (build 4) I have added the size in log as this example:

[LOG] Upload complete https://nextcloud.xxxxxxxxx.org/remote.php/webdav/0/20-09-02 10-54-07 0015.jpg of size: 91107

**Comment by Subito** [comment]
For me its only reproducable on a low-latency Connection. Unfortunatly I don't have access to one right now but I will test the TestFlight Version once I have.

**Comment by marinofaggiana** [comment]
Nextcloud TestFlight

https://testflight.apple.com/join/GjNbfo2a

**Comment by marinofaggiana** [comment]
I have find a possible issue please test with the last 3.0.7 (build 6)

**Comment by marinofaggiana** [comment]
I close, reopen if happen yet

**Comment by ChildLearningClub** [comment]
I just commented [here](https://github.com/nextcloud/ios/issues/2061) about finding corrupted video files that have shown to have successfully uploaded to the server. 

`Nextcloud 27.1.1 iOS 4.8.6.4` 

I'm not sure this should be closed until adequate checks are put into place to verify that uploaded files are matching copies of those from the device. 

There are videos that I synced the other day that are not playable on the server. The vides are still on the phone and play fine and have all their metadata, but the corrupted ones that where uploaded through the iOS app lack much of the item properties and GPS data. 

So then we have a corrupted file on the server that without checking or seeing that the thumbnail has not generated may not know that it is corrupted and safely assume that we can delete the copy off the phone. 

Could not something like the [ExifTool by Phil Harvey](https://exiftool.org/) be implemented to do a final check of both files to verify they are exact matches? If not, prevent the corrupted data from being stored on the server? Continue to retry the upload sync and if it still can not be done properly leave it in the upload sync or have a very clear display to the user of objects that could not be synced correctly?

Anyway, I feel this is an essential part of having a sync client.
